rclcpp/include/rclcpp/any_subscription_callback.hpp

```diff
@@ -487,9 +487,13 @@ class AnySubscriptionCallback
   }
 
   // Dispatch when input is a ros message and the output could be anything.
-  void
+  template<typename MaybeConstROSMessageType>
+  typename std::enable_if<std::is_same<ROSMessageType,
+    typename std::remove_const<MaybeConstROSMessageType>::type
+    >::value
+  >::type
   dispatch(
```

Comment:
could you add tests making sure that loaned messages work well with all types of callbacks?
There're a lot of codepaths here, and it's pretty hard to make sure you didn't break anything else.
So it would be nice to make sure all cases are covered in tests (some of the cases might already be covered).